### ISSUE TITLE
License declaration slightly ambiguous

### ISSUE THREAD

**Issue by jengelh** [openucx/ucx#302]
Many files of the ucx source, pick for example `src/ucp/api/ucp.h`, read:
```
/**
* Copyright (C) Mellanox Technologies Ltd. 2001-2014.  ALL RIGHTS RESERVED.
* $COPYRIGHT$
* $HEADER$
*/
```
A file which declares only all-rights-reserved (ARR) does not permit distribution, use, nor modification. I presume the intent here was to substitute $HEADER$ by the text of the "README" file, but as this has not happened, a recipient of the file is left wondering. I recommend to remove the comment block (in which case the top-level LICENSE clearly applies), or, as a second option, to do the substitution at the expense of source bloat.

**Comment by yosefe** [comment]
@shamisp

**Comment by shamisp** [comment]
Thanks for the comment, I will take a look later today. In any case the code is distributed under bsd license as described at www.openucx.org

**Comment by shamisp** [comment]
Also, it is important to note that all the contributors of the project have to sign the contributor agreement (www.openucx.org) that explicitly states that code is submitted under BSD3 license.  IMHO, the license is pretty clear.

I don't like the idea of removal of the copyright lines, but we may consider to include entire license in each file.

**Comment by jengelh** [comment]
You can also include them by reference, e.g. "see file xyz.txt for terms"

**Comment by shamisp** [comment]
"See file LICENSE for terms" - I like it. We can replace $COPYRIGHT$, $HEADER$ with the proposed line.

**Comment by shamisp** [comment]
@jengelh - Brief update: I'm delaying the change since we have an upcoming code restructuring for ugni transport and I would like to prevent potential conflicts which can be introduced by the license update. Once we done with the restructuring we will update the license.

**Comment by shamisp** [comment]
The issue was addressed in #375
